src/query.js

```diff
@@ -31,6 +32,70 @@ function toQueryString(obj) {
   return qs;
 }
 
+const findStylePreset = (id, stylePresets) => {
+  const flatPresets = [
+    ...stylePresets.filter(p => p.type !== 'sublist'),
+    ...stylePresets.filter(p => p.type === 'sublist').flatMap(p => p.presets),
+  ];
+
+  return flatPresets.find(preset => preset.id === id);
+};
+
+const findBranchPattern = (map, branchPatterns) => {
+  const branchPattern = branchPatterns.find(p => p.id === map.branchPatternId);
+  if (!branchPattern) return null;
+  return {
+    ...branchPattern,
+    url: createBranchUrl(
+      branchPattern.pattern,
+      map.branch,
+      map.branchPatternStyle
+    ),
+  };
+};
+
+const encodeMaps = (maps, config) => {
+  const mapsToEncode = maps.map(m => {
+    let encodeMap = { id: m.id };
+    let configMap;
+
+    if (m.branchPatternId) {
+      configMap = findBranchPattern(m, config.branchPatterns);
+      encodeMap = {
+        branchPatternId: configMap.id,
+        branchPatternStyle: m.branchPatternStyle,
+        branch: m.branch,
+      };
+    } else {
+      configMap = findStylePreset(m.id, config.stylePresets);
+    }
+
+    if (!configMap) return m;
```

Comment:
We need a condition for custom urls since there's no preset for it in the config. It's defaulting here to just return the whole map object which contains a `style` property with the entire stylesheet and that's getting into the URL. I think with just the URL, id, and renderer it should work as expected.